server/server/src/discovery.rs

```diff
@@ -1,39 +1,70 @@
+#[cfg(target_os = "windows")]
+use {astro_dnssd::DNSServiceBuilder, std::collections::HashMap};
 
 const SERVICE_NAME: &'static str = "_omsupply._tcp";
 const NAME: &'static str = "omSupplyServer";
-
 const PROTOCOL_KEY: &'static str = "protocol";
 const CLIENT_VERSION_KEY: &'static str = "client_version";
 const HARDWARE_ID_KEY: &'static str = "hardware_id";
-
 const CLIENT_VERSION: &'static str = "unspecified";
 
 pub(crate) fn start_discovery(protocol: Protocol, port: u16, hardware_id: String) {
     tokio::task::spawn(async move {
-        let mut txt: TxtRecord = TxtRecord::new();
-        txt.set_value(HARDWARE_ID_KEY.as_bytes(), hardware_id.as_bytes())
-            .unwrap();
-        txt.set_value(CLIENT_VERSION_KEY.as_bytes(), CLIENT_VERSION.as_bytes())
-            .unwrap();
-        txt.set_value(PROTOCOL_KEY.as_bytes(), protocol.to_string().as_bytes())
+        #[cfg(any(target_os = "macos", target_os = "linux"))]
+        {
+            let mut txt: TxtRecord = TxtRecord::new();
+            txt.set_value(HARDWARE_ID_KEY.as_bytes(), hardware_id.as_bytes())
+                .unwrap();
+            txt.set_value(CLIENT_VERSION_KEY.as_bytes(), CLIENT_VERSION.as_bytes())
+                .unwrap();
+            txt.set_value(PROTOCOL_KEY.as_bytes(), protocol.to_string().as_bytes())
+                .unwrap();
+            let (_registration, _) = async_dnssd::register_extended(
+                SERVICE_NAME,
+                port,
+                RegisterData {
+                    txt: txt.rdata(),
+                    name: Some(NAME),
+                    ..Default::default()
+                },
+            )
+            .unwrap()
+            .await
             .unwrap();
 
-        let (_registration, _) = async_dnssd::register_extended(
-            SERVICE_NAME,
-            port,
-            RegisterData {
-                txt: txt.rdata(),
-                name: Some(NAME),
-                ..Default::default()
-            },
-        )
-        .unwrap()
-        .await
-        .unwrap();
+            // Without this discovery stops (even if result of register_extended is kept and passed to caller)
+            futures::future::pending::<()>().await;
+        }
 
-        // Without this discovery stops (even if result of register_extended is kept and passed to caller)
-        futures::future::pending::<()>().await;
+        #[cfg(target_os = "windows")]
+        {
+            let mut text_record = HashMap::<String, String>::new();
+            text_record.insert(HARDWARE_ID_KEY.to_string(), hardware_id.to_string());
+            text_record.insert(CLIENT_VERSION_KEY.to_string(), CLIENT_VERSION.to_string());
+            text_record.insert(PROTOCOL_KEY.to_string(), protocol.to_string());
+            // need to keep the thread running
+            // found that the discovery client in electron did not pick up the server at times
+            // and running register again was necessary for the server to be found
```

Comment:
Would `futures::future::pending::<()>().await;` work ? so instead of loop we just have that at the end

Reply:
it didn't work for me - as well as the further issue of the discovery wanting a regular restart to keep it working 🤷